keycloak-service-providers/magic-link/src/main/java/org/kindlyops/auth/magic/MagicLinkFormAuthenticator.java

```diff
@@ -54,8 +55,17 @@ public void action(AuthenticationFlowContext context) {
 
         String key = KeycloakModelUtils.generateId();
         context.getAuthenticationSession().setAuthNote("email-key", key);
+        String link = "";
+        if (sessionId == "none") {
```

Comment:
since line 57 is calling `context.getAuthenticationSession()`, I wonder whether we could directly call `context.getAuthenticationSession().getId()` rather than using rootAuthenticationSession?

What is RootAuthenticationSession used for in other parts of the code? Is it like the root acount?

Reply:
I tracked down this documenation for `createRootAuthenticationSession()` in the keycloak code: 
> Creates and registers a new authentication session with random ID. Authentication session entity will be prefilled with current timestamp, the given realm and client.
So I think this usage is correct, it's not like using a root login, it means root like "base".